pkg-config, pkgconf: Misc fixes

Ericson2314 opened the pull request:
###### Motivation for this change

See each commit for details. Most importantly this fixes builds using pkg-config-provided macros, like `libarchive`.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Ericson2314 committed:
pkg-config: Document new location of bug

Ericson2314 committed:
pkg-config: If building vanilla version, can run original tests

Ericson2314 committed:
pkg-config: Multiple outputs

This is good practice, and will help with tests. Wrapper adjust to match
bintools-wrapper re these new outputs.

Ericson2314 committed:
pkg-config-wrapper: Provide autoconf macros

This fixes libarchive, and probably numerous other builds.

bhipple commented on pkgs/build-support/pkg-config-wrapper/default.nix:
```diff
@@ -87,13 +96,15 @@ stdenv.mkDerivation {
       ##
-      ## Man page and info support
+      ## Man page and doc support
       ##
```
Put the comment outside the builder so it doesn't cause a rebuild in the future, perhaps?

Ericson2314 replied:
Indeed! And at some point somebody should do the same for `cc-wrapper` and `bintools-wrapper`---I put the `##` in those before I learned to regret it.

Ericson2314 committed:
pkgconf: Multiple outputs

Ericson2314 committed:
pkgconf: Wrap with pkg-config-wrapper

Ericson2314 committed:
pkg-config-wrapper: Move comments outside of strings

This is much better because then we can freely keep the comments up to
date without causing mass rebuilds.

Someday, somebody should make the same change with `cc-wrapper` and
`bintools-wrapper`.

Ericson2314 commented:
OK not sure why that one ofborg job is stalled, I built everything locally.
